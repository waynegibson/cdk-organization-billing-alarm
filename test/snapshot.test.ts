import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import {
  MasterAccountAlarmTest,
  MasterAccountAlarmAssociatedWithMultipleEmailAddressToAwsServiceTest,
  SubscribeMultipleEmailAddressToExistingTopic,
  MultipleLinkedAccountsAssociatedToExistingTopicTest,
} from '../src/integ.default';

test('integration snapshot test to create a master account billing alarm with a new sns topic', ()=> {
  const integration = new MasterAccountAlarmTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test to create a master account billing alarm with multiple email addresses, associated with a aws service', ()=> {
  const integration = new MasterAccountAlarmAssociatedWithMultipleEmailAddressToAwsServiceTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test to subscribe multiple email addresses to existing topic', ()=> {
  const integration = new SubscribeMultipleEmailAddressToExistingTopic();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test to link multiple linked account billing alarms to an existing topic', ()=> {
  const integration = new MultipleLinkedAccountsAssociatedToExistingTopicTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});