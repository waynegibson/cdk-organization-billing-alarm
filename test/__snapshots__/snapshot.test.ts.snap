// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration snapshot test to create a master account billing alarm with a new sns topic 1`] = `
Object {
  "Outputs": Object {
    "MasterAccountAlarmAccountBillingAlarmTopicArn4B78C86B": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD",
      },
    },
  },
  "Resources": Object {
    "MasterAccountAlarmAccountBillingAlarm98422C04": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD",
          },
        ],
        "AlarmDescription": "Consolidated billing alarm for all service charges",
        "AlarmName": "Consolidated: (All Services)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonDynamoDB",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 140,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD": Object {
      "Properties": Object {
        "DisplayName": "Organizational billing alarm topic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "MasterAccountAlarmAccountBillingAlarmTopicadminexampleorgE8847B2C": Object {
      "Properties": Object {
        "Endpoint": "admin@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test to create a master account billing alarm with multiple email addresses, associated with a aws service 1`] = `
Object {
  "Outputs": Object {
    "MasterAccountAlarmAccountBillingAlarmTopicArn4B78C86B": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD",
      },
    },
  },
  "Resources": Object {
    "MasterAccountAlarmAccountBillingAlarm98422C04": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD",
          },
        ],
        "AlarmDescription": "Consolidated billing alarm for Amazon DynamoDB charges",
        "AlarmName": "Consolidated: (Amazon DynamoDB)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonDynamoDB",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 140,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD": Object {
      "Properties": Object {
        "DisplayName": "Organizational billing alarm topic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "MasterAccountAlarmAccountBillingAlarmTopicadminexampleorgE8847B2C": Object {
      "Properties": Object {
        "Endpoint": "admin@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "MasterAccountAlarmAccountBillingAlarmTopicbillingexampleorgD794BB99": Object {
      "Properties": Object {
        "Endpoint": "billing@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "MasterAccountAlarmAccountBillingAlarmTopic1BDD26AD",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test to link multiple linked account billing alarms to an existing topic 1`] = `
Object {
  "Outputs": Object {
    "LinkedAccountAlarm165881TopicArn627F4D55": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "{{resolve:secretsmanager:arn:",
            Object {
              "Ref": "AWS::Partition",
            },
            ":secretsmanager:",
            Object {
              "Ref": "AWS::Region",
            },
            ":",
            Object {
              "Ref": "AWS::AccountId",
            },
            ":secret:test/billing/topicArn:SecretString:::}}",
          ],
        ],
      },
    },
    "LinkedAccountAlarm5ecf13TopicArn1B7F164C": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "{{resolve:secretsmanager:arn:",
            Object {
              "Ref": "AWS::Partition",
            },
            ":secretsmanager:",
            Object {
              "Ref": "AWS::Region",
            },
            ":",
            Object {
              "Ref": "AWS::AccountId",
            },
            ":secret:test/billing/topicArn:SecretString:::}}",
          ],
        ],
      },
    },
    "LinkedAccountAlarmf2b42bTopicArnD8EF25B4": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "{{resolve:secretsmanager:arn:",
            Object {
              "Ref": "AWS::Partition",
            },
            ":secretsmanager:",
            Object {
              "Ref": "AWS::Region",
            },
            ":",
            Object {
              "Ref": "AWS::AccountId",
            },
            ":secret:test/billing/topicArn:SecretString:::}}",
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "LinkedAccountAlarm165881Alarm25F0FE5B": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "{{resolve:secretsmanager:arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":secretsmanager:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":secret:test/billing/topicArn:SecretString:::}}",
              ],
            ],
          },
        ],
        "AlarmDescription": "Billing Alarm for AWS DynamoDB charge estimates only (Account: 123456789000)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "LinkedAccount",
            "Value": "123456789000",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonDynamoDB",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 120,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "LinkedAccountAlarm5ecf13AlarmB6C702E9": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "{{resolve:secretsmanager:arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":secretsmanager:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":secret:test/billing/topicArn:SecretString:::}}",
              ],
            ],
          },
        ],
        "AlarmDescription": "Consolidated billing alarm: All AWS service charge estimates (Account: 123456789000)",
        "AlarmName": "Consolidated (123456789000): AWS Services",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "LinkedAccount",
            "Value": "123456789000",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 50,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "LinkedAccountAlarmf2b42bAlarm64FB5A98": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "{{resolve:secretsmanager:arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":secretsmanager:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":secret:test/billing/topicArn:SecretString:::}}",
              ],
            ],
          },
        ],
        "AlarmDescription": "Consolidated billing alarm for all AWS service charge estimates",
        "AlarmName": "Consolidated (444455556666): AWS Services",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "LinkedAccount",
            "Value": "444455556666",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 50,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
  },
}
`;

exports[`integration snapshot test to subscribe multiple email addresses to existing topic 1`] = `
Object {
  "Resources": Object {
    "EmailSubscriptionTopicadminexampleorg8727E587": Object {
      "Properties": Object {
        "Endpoint": "admin@example.org",
        "Protocol": "email",
        "TopicArn": "{{resolve:ssm:/test/billing/topicArn:1}}",
      },
      "Type": "AWS::SNS::Subscription",
    },
    "EmailSubscriptionTopicbillingexampleorg72D7268D": Object {
      "Properties": Object {
        "Endpoint": "billing@example.org",
        "Protocol": "email",
        "TopicArn": "{{resolve:ssm:/test/billing/topicArn:1}}",
      },
      "Type": "AWS::SNS::Subscription",
    },
    "EmailSubscriptionTopicjohnexampleorgCF4D0E86": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": "{{resolve:ssm:/test/billing/topicArn:1}}",
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;
